### PR TITLE
map NotImpmentedError to frontend

### PR DESCRIPTION
**Pull request by MollyZhang**
solving issue #176 , map NotImplementedError to frontend exceptions to communicate the error to client.

### CODE REVIEW

File: ga4gh/frontend_exceptions.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import exceptions
```
Comment (dcolligan) [inline ga4gh/frontend_exceptions.py]: isn't this module imported automatically?

Comment (dcolligan) [inline ga4gh/frontend_exceptions.py]: https://docs.python.org/2/library/exceptions.html

"The exceptions are defined in the module exceptions. This module never needs to be imported explicitly: the exceptions are provided in the built-in namespace"

**Comment by MollyZhang** [comment]
@dcolligan  updated

File: ga4gh/frontend_exceptions.py [hunk 2]
```diff
@@ -112,6 +112,14 @@ def __init__(self):
         self.code = 10
 
 
+class NotImplementedException(FrontendException):
```
Comment (dcolligan) [inline ga4gh/frontend_exceptions.py]: This should actually inherit from ServerException

File: ga4gh/frontend_exceptions.py [hunk 3]
```diff
@@ -112,6 +112,14 @@ def __init__(self):
...
+
+    def __init__(self):
+        super(FrontendException, self).__init__()
+        self.httpStatus = 500
```
Comment (dcolligan) [inline ga4gh/frontend_exceptions.py]: get rid of this line, since it will inherit the httpStatus from its parent

**Comment by MollyZhang** [comment]
@dcolligan  Okay I changed those

File: ga4gh/frontend_exceptions.py [hunk 4]
```diff
@@ -112,6 +112,13 @@ def __init__(self):
         self.code = 10
 
 
+class NotImplementedException(ServerException):
+
+    def __init__(self):
+        super(FrontendException, self).__init__()
```
Comment (dcolligan) [inline ga4gh/frontend_exceptions.py]: FrontendException --> NotImplementedException

**Comment by MollyZhang** [comment]
@dcolligan  updated

**Comment by dcolligan** [comment]
+1

**Comment by jeromekelleher** [comment]
+1.

**Pull request merged**